### ISSUE TITLE
General UI Issues

### ISSUE THREAD

**Issue by aberfan** [JohnnyRS/PandaCrazy-Max#31]
In case these aren't known:

- Renaming tabs sometimes doesn't take. No error, but on reload they're back to a previous name. Workaround is exporting and then importing back.

- Can't mute individual jobs.

- Dialog for deleting, for jobs or queue, requires 3 keys now. Focus should be on "Yes" to allow optimal behavior. Really annoying clicking on "Return" and then having to tab three times and finally hit enter, when before it was just click, enter, to delete.

- Queue doesn't seem to get updated when you return a hit, but instead seems to wait for whatever duration queue update timer is.

- Developer console log is nearly impossible to read with dark theme developer mode enabled.

- Moving jobs between tabs sometimes have the pop-up about the job "stick" so it stays on the screen, with the only way to get rid of it being reloading.

- String "Total Earnings" is inaccurate, and should be the previous "Total Accepted"

**Comment by JohnnyRS** [comment]
Thanks for letting me know about the tabs. Will have to look at it to see what is happening. I don't use tabs regularly so didn't know.

Clicking return should delete automatically in the delete dialog. I'll have to recheck it or make sure the yes if focussed. Could be the difference between computers too. Not sure. Either way I did find another bug with the deleting jobs that I need to fix.

Returning a hit needs to wait for a success message from mturk before it removes it so I decided to just let it recheck the queue instead. Maybe I'll have it pause with a wait message or figure out a better way.

The developer debug stuff needs some work before It's finished. Had a lot of debugging ideas at the beginning but it was too confusing so need to be reworked.

Total earnings is supposed to be the earning given by mturk dashboard plus the hits in queue. But right now it only works with the mturk suite extension installed. Later I am hoping to have the script calculate the earnings on it's own. I think potential earnings would be more accurate but that is too long.

I'll have to recheck the moving jobs around. It's always been a weird buggy thing to use.

**Comment by aberfan** [comment]
For the return, I think it is just  a need to focus 'Yes".

For the queue bit, I understand the reasoning there. If you just removed it from PC and it wasn't actually removed, it'd be weird to see it suddenly come back on the next refresh.  The way it is now, though, if you have a longer refresh (which I do) is if you're trying to go through and delete a lot of hits, you can't just do it from the top one, because you might have that and the next one be ones you've already removed.  For some sort of compromise maybe just have a task you've removed in a faded out color and disabled, so there's some sort of visual feedback it's "gone"?  That would work for me, because if it's "there but not there" I'd just see the first one that's clickable.  If for some reason it fails to remove, then the first one would be enabled again, and maybe some message about errors?

Debugging was just a note; I ended up just switching dev console to light mode to read, but unless you really care about it, it might make sense just not to have colors there. I had the background page open because I wanted to see if I could give you more information about what was happening with the overlay hanging around when moving jobs between the tabs and not saving the renames.

Tabs, btw, are one of my favorite parts of PC.  It's nice to dump things from requesters or type of hits into one and not clutter up the main tab with stuff you want to keep but not have in your face (Like Geohives off the top of my head), so it's great you did them even though you don't use them heh!

If the total earnings is supposed to be from mturk, I'd put a feature request here to allow disabling that and instead tracking what PC's collected as in the old behavior. I don't want to install MTS, and I don't particularly want my hit grabber to display information from my dashboard; I go to my dashboard, when I want to see it. Again I'm an outlier, but it tends to motivate me better that way and I was annoyed when AMT forced that stupid goal bar on us whether we wanted it or not.  

I think with all these things I'm asking maybe I should learn to code so I can help out or something.  Thanks again for listening!

**Comment by JohnnyRS** [comment]
The new update should have the delete dialog use return key for the yes button.
Renaming tabs should now save the new name.

The rest I'll be checking out for future updates.

**Comment by aberfan** [comment]
Nice! Thanks for the update!

Some notes from using this:
+ There's some bug with that flashing thing. I can't give a way of replicating yet, but it definitely exists: Run a job for awhile, and turn off collecting, and the GID continues blinking, and nothing stops it short of reloading. (That is, switching tabs, starting and stopping that job again, or others, etc).  This doesn't happen all the time, even for the same GID, so I'm not sure what the commonality is.

+ Total Earnings does nothing. You said this only works with MTS installed; maybe have it fall back to "Total Accepted" if MTS isn't installed? (I think MTS had something that alerted it was there; not sure if this is detectable?).  I've noticed that sometimes this gets a value for awhile and then goes back to $0, maybe reflecting what's in the queue.

+ Another gui note is that maxed out tasks stay red 'for a while' after you turn them off, but then eventually return to normal.

And not sure if there's a bug or something, but I lost out of a decent task today, because the job a search created seems to have only ran for a short duration.  I'm not sure if that's just me missing a setting: I want any job created by a search to run until I turn it off. Also, this feature is absolutely brilliant, and I wanted to mention that to you.  I typically use this for UNU, and it lends itself perfectly for that requester!  (I'm using the same interface, not the new search whatever thing.)

Also I'm not sure if this is one of the things fixed with alarms in the latest, but when I'm logged out, I never heard the logged out sound. I use custom sounds, and can play all of them in the alarm settings. (I like that now the "you're not logged in" dialog goes away on its on, instead of me having to click it; that's a nice upgrade from the old way.)

**Comment by JohnnyRS** [comment]
You can now mute jobs by clicking on the mute button in the details dialog or right clicking on the panda job. Should give you a dotted border when a job is muted.

The bug with the flashing groupid is because the window was minimized or you went to a different tab. Chrome stops all animations when that happens but not the timing like the original script. I stopped the animation when the script is not focussed instead of giving a warning now so it should be fixed.

The red color happens when there is a limit for that job. If you turn off the job it will still show that the job is still limited so you won't be confused why it's not doing anything if you turn it back on again. If the red color turns off then the limit is released so turning it back on will start collecting again. That is by design.

Any job created by a search/trigger will always have a duration set by the search/trigger options. You can change that in the options by changing the stop collecting options in the search/trigger job. Also manually stopping a panda job and turning it back on should let it collect until you stop it. I set the duration for auto collecting jobs at 12 seconds because panda jobs share the timer with all other panda jobs but all search jobs get checked every second. If it didn't collect it in that 12 seconds it's probably already collected by others usually. The search/trigger job will continue searching for any others or returned hits.

The logged off alarm should now work correctly.

**Comment by aberfan** [comment]
Thank you for the explanations and the fixes.

Regarding the search trigger, does having "Stop collecting after (seconds)" at 0 now work to stop it from automatically stopping after 12 seconds? It's unfortunate when the created panda stops actually collecting at all; my usage, for example with UNU, it can still grab within a few minutes to the end of the hour, and there's also surveys where people toss back, certainly after 12 seconds. 

One random other UI issue: Lower window doesn't remember its position across reloads, and sometimes the divider goes almost completely to the bottom of the screen. 

Thanks again for addressing bugs and being receptive to feedback; I appreciate your hard work and efforts!

**Comment by JohnnyRS** [comment]
The stop collecting option should be above 0. That's a bug. No automation from search jobs should let a panda job collect with no limits. The user is the only one who should make that decision. So the way to allow it to collect for an hour is to put in 3600 for the seconds. Right now there is a bug that doesn't set the min or the max to some of those options. You can also turn off the panda job and then turn it back on.

I'll try to explain the reason why. Let's say you have 5 panda jobs running and then a well paid survey drops. If you have a search job/trigger on it will create a panda job which makes the script have 6 panda jobs. If there was no restrictions this is what would happen. Let's assume search timer is at 1000 ms. Panda timer is at 1000 ms too. So every 6 seconds the script tries to collect that survey. Panda jobs are more likely to grab it than a search job but they all have to take turns. Search job/trigger is always on and do not take turns so will be checked every second. If a search job finds that someone returned that well paid survey then it will create or start the panda job for that survey which should take less than a second. The panda job should be able to pick it up if you're lucky. So a search job/trigger tries to collect a hit every 1 to a max 2 seconds if it finds it. Compare that to 6 seconds if 6 panda jobs are collecting at once. Obviously if you only have the survey panda running than the chance of it picking it up is a lot higher than a search job.

So I placed a limit for the collecting panda so users won't find themselves with dozens of panda's running all at once when they didn't realize hits were being created. There will be options in the next update for the default values for search triggers.

Search group id triggers only will disable if you have it set to accept only once and it collects that hit. I know I really need to write all this in the wiki so it can be understood but that's for later. The search jobs are very different now compared to the original script and needs to get used to.

Later I'll deal with saving the lower panels but most don't use it from my experience.

**Comment by aberfan** [comment]
Thank you for the comprehensive response. So from what you're saying, I just need to set the "Stop Collecting" to 3600 or whatever and it'll work properly?  Trying to change the 'Stop collecting after (seconds)' doesn't seem to allow a value higher than 120.

For the rest, what I'm understanding you saying is that because you don't trust users to know what they want, you're doing non-intuitive behavior which is going to cause a lot of people frustration, because instead of doing what's expected - "Run until I tell it not to" - it's "Run an arbitrary amount of time, that's set in some setting, that's not well documented".   Please note I'm not trying to be insulting and I'm not sure how to type this in a way that doesn't sound a bit like it is.

For me, in particular, I run a single search , and when it finds the GID, I usually turn off the search.  For a given requester, like say Remesh, you'd want 2-3 pandas going simultaneously if for some reason those all dropped at once.  For something like Wharton you might not but again, it's simple enough for someone to turn things off they don't need.

I hear where you're coming from but someone with 5 panda jobs running already has lowered chances by 1/5 of grabbing something. Making it 1/6th isn't really changing outcomes that much.  

tl;dr It's confusing that a GID is automatically created (awesome!) and then turns off after an arbitrary amount of time that's not globally settable (bummer!).   But if I can just put the handful of search triggers I use to 3600 I'm happy, it'll just bite other people who haven't read this or the wiki lol!

(Regarding the "you can also turn off the panda job and then turn it back on", if the search trigger finds a hit I want in the 2-3 minutes I'm making another pot of coffee, then it'll have turned off in the optimal first minute, which is really really bad. It's my experience that the first 60 seconds are the most fruitful.)

Lower panel is just a general UI issue; any time the queue goes over 15+ tasks, you can't see them, unless you expand it, so everyone expands it. Without it saving, any time you resize the window or reload, you have to grab the tiny triangle in the lower right and resize again; it was easier to resize when it was a grab-able horizontal bar, than now, which is why it'd be nice if it remembered the position.

**Comment by JohnnyRS** [comment]
Correct setting it to 3600 will work but the min and max is broken so that will be fixed in the next update.

Actually it's the opposite from my experience. I see and get many questions about why they aren't getting hits when they have 50 panda jobs running. I also see many questions about the color meanings but that is in the help page. So new users usually won't read instructions and I have to try to keep them from making mistakes. I actually trust the user more than the script to make decisions and that is why anything the script does automatically has limits. Anything manually done will only have limits set by the user. I just have to figure out how to explain it better and try to put that explanation where the user will see it.

You really should try running many searches. I have 100 search triggers enabled and only turn off the low timed hits when I'm away. Unu is always on because those hits won't overlap. C-sats is on too but I limit the group ID in queue to only 2 hits so each panda will only accept 2 hits each which stops the queue from being filled with hits I won't be able to finish. Those 100 search jobs will be checked in a second so there is no reason to disable them. Haven't tested what the limit would be but my guess would be 1000 search triggers should do it in a second too. Then when custom searches get completely finished it should really help a lot.

**Comment by aberfan** [comment]
> Correct setting it to 3600 will work but the min and max is broken so that will be fixed in the next update.

Great! ;)

> Actually it's the opposite from my experience. I see and get many questions about why they aren't getting hits when they have 50 panda jobs running.
> So new users usually won't read instructions and I have to try to keep them from making mistakes.

This way of thinking seems so alien to me, because you can never know what is a mistake for someone else or not. If someone isn't going to bother to read the help, why invest more effort for them, rather than people who care enough to? I'll note I've spent a lot of quality time on your website reading the docs. :-)

> I actually trust the user more than the script to make decisions and that is why anything the script does automatically has limits.

This seems to contradict itself; if you trust the user, then why is the script doing things automatically?  My thinking here is "I set this thing to do something, and it did it! Yay! Wait, what? Why'd it stop doing it?!!! I didn't ask it to stop! Why does this think it knows better than me what I want? Maybe I want to have 50 pandas going at once! Do what I say unless I say do something else!"

Of course, this is just me sharing my opinion, which doesn't matter.  When the min/max is fixed, I'll just set to 3600 or higher where appropriate, and I'll be set.  That'll take care of long running things like IPS as well as the 3+ hour timer Gershwins and the like.  Just for what it's worth, know at least one user thinks it's really non-intuitive to have something properly start and then stop without intervention or say-so.

> You really should try running many searches. I have 100 search triggers enabled and only turn off the low timed hits when I'm away. Unu is always on because those hits won't overlap. C-sats is on too but I limit the group ID in queue to only 2 hits so each 

Ah, C-SATs, how I miss them.  I don't have 100 things I care enough to search for.  In fact, there are only 5 or so requesters who I'll have a search trigger going for, ever.  For the most part, I just use a notification to tell me when requesters I do tasks for drop, and I manually start a limited panda (that is, [x] in queue at once) for them.   There are two requesters that have tasks at a certain time each (most) days, that I just turn the panda on slightly before, and off afterwards, because otherwise, there's no reason to have them going.  It just comes down to different folks having different workflows.

Thank you again for making such a useful and powerful extension, and giving away your efforts as you have, and I hope you don't take it wrong, me expressing a contrary opinion.
